File: January/Day7/leetcode-134.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        
        int n = gas.size();
        int totalgas = 0 , totalcost = 0;
        int currentgas = 0;
        int lowest_Index = 0;

        for(int i = 0; i < n; i++)
        {
            totalgas += gas[i];
            totalcost += cost[i];
            currentgas = currentgas + gas[i] - cost[i];

            if(currentgas < 0)
            {
                lowest_Index = i+1;
                currentgas = 0;
            }
        }

        if(totalgas >= totalcost)
        {
            return lowest_Index;
        }
        return -1;
    }
};

File: January/Day7/leetcode_134.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int gascost=0;
        int actualcost=0;
        int index=0;
        int total=0;
        for(int i=0;i<gas.length;i++){
            gascost+=gas[i];
            actualcost+=cost[i];
            total+=gas[i]-cost[i];
            if(total<0){
                index=i+1;
                total=0;
            }

        }
        return gascost<actualcost?-1:index;
    }
}
